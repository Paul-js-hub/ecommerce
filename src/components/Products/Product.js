import { Card, Col } from "antd";
import { Link } from "react-router-dom";
const { Meta } = Card;
const Product = ({product}) => {
  return (
    <div>
      <Col>
        <Link to={`/product/${product.id}`}
        state={{product}}
        >
        <Card
          hoverable
          style={{
            width: 240,
          }}
          cover={
            <img
              alt="product"
              src={product.image}
              style={{
                width: 240,
                height: 240
              }}
            />
          }
        >
          <Meta title={product.title} description={`${product.description.substring(0, 21)}...`}  quantity={product.quantity}/>
          <p>{`$ ${product.price}`}</p>
        </Card>
        </Link>
      </Col>
    </div>
  );
};

export default Product;
