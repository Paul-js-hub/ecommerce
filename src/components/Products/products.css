.products-container {
    overflow-x: hidden !important;
}