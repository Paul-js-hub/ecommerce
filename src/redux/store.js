import { configureStore } from '@reduxjs/toolkit';
import productsReducer from './products/products.js';
import cartsReducer from './cart/cart.js';

const store = configureStore({
  reducer: {
    products: productsReducer,
    cart: cartsReducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({ serializableCheck: false})
});

export default store;
