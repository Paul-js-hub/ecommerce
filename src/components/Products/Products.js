import { useSelector, useDispatch } from "react-redux";
import { useEffect } from "react";
import { getProducts } from "../../redux/products/products";
import Product from "./Product";
import { Row } from "antd";

const Products = () => {
  const dispatch = useDispatch();
  const result = useSelector((state) => state.products);
  useEffect(() => {
    dispatch(getProducts());
  }, [dispatch]);

  return (
    <Row
      gutter={{
        xs: 8,
        sm: 16,
        md: 24,
        lg: 32,
      }}
    >
      {result.products.map((product) => {
        return (
          <Product
            key={product.id}
            id={product.id}
            title={product.title}
            description={product.description}
            image={product.image}
            price={product.price}
          />
        );
      })}
    </Row>
  );
};

export default Products;
