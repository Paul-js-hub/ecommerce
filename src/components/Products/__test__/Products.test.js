import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../../redux/store'
import Products from '../Products'

describe("Test Products Component", () => {
  it('renders all products', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Products />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})