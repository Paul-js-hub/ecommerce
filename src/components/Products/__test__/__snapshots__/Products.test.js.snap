// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Products Component renders all products 1`] = `
<div
  className="products-container"
>
  <div
    className="ant-row css-dev-only-do-not-override-9ntgx0"
    style={
      Object {
        "marginLeft": -16,
        "marginRight": -16,
      }
    }
  >
    <div
      className="spinner-container spinner-border"
      role="status"
    >
      <span
        className="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
